solr/core/src/java/org/apache/solr/handler/component/AffinityReplicaListTransformerFactory.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.solr.handler.component;
+
+import org.apache.solr.common.params.CommonParams;
+import org.apache.solr.common.params.ShardParams;
+import org.apache.solr.common.util.NamedList;
+import org.apache.solr.request.SolrQueryRequest;
+
+/**
+ *
```

Comment:
`AffinityReplicaListTransformer` being public but `AffinityReplicaListTransformerFactory` here not being public jumps out as surprising, I would have expected them both to have the same visibility or potentially for the factory to have more visibility that the class itself. And perhaps the javadocs could mention e.g. about the defaulting-to-Q-param behaviour or so. What do you think?

Reply:
Yes, that makes sense. I'll make `AffinityReplicaListTransformer` package-private (like `AffinityReplicaListTransformerFactory`, and `ShufflingReplicaListTransformer`). I'll flesh out the javadocs too.